PR: https://github.com/odoo/odoo/pull/29308

From: 66f0e26f6f3f2155fcfcc55e84f2e76cbfdb2c1d
From: Xavier Morel
Date: 2018-12-07 13:37:34

Structural Changes: 6
Total Changes: 6

[CHG] *: make binary fields default to attachment=True

Check that it makes custom binary fields into attachment as that's the
main reason for the change: when users create binary fields via Studio,
they're necessarily db-stored (as the interface doesn't allow altering
the attachment attribute and it's unclear how we'd handle users
switching it on/off every time), which significantly bloats their
database (and burns storage & backup space), especially as the primary
use case for binary fields is adding images and documents to records.

* check that binary fields are properly created as attachment=True
* add attachment=False on fields where that seems relevant (most but not
  all of the fields previously using the default)
* remove occurrences of attachment=True

closes odoo/odoo#29308

================================= pseudo patch: =================================

--- a/addons/point_of_sale/models/pos_category.py
+++ b/addons/point_of_sale/models/pos_category.py
@@ -20,13 +20,13 @@ class PosCategory(models.Model):
     # thumbnails for categories. However if we have a thumbnail for at least one
     # category, then we display a default image on the other, so that the
     # buttons have consistent styling.
-    image = fields.Binary(attachment=True,
+    image = fields.Binary(
         help="This field holds the image used as image for the cateogry, limited to 1024x1024px.")
-    image_medium = fields.Binary(string="Medium-sized image", attachment=True,
+    image_medium = fields.Binary(string="Medium-sized image",
         help="Medium-sized image of the category. It is automatically "
              "resized as a 128x128px image, with aspect ratio preserved. "
              "Use this field in form views or some kanban views.")
-    image_small = fields.Binary(string="Small-sized image", attachment=True,
+    image_small = fields.Binary(string="Small-sized image",
         help="Small-sized image of the category. It is automatically "
              "resized as a 64x64px image, with aspect ratio preserved. "
              "Use this field anywhere a small image is required.")
